src/tui/rendering.rs

```diff
@@ -99,16 +101,25 @@ impl Tui {
         self.search_area.insert_char(new_char);
     }
 
-    pub fn process_details_down(&mut self) {
-        self.process_details_scroll = self.process_details_scroll.saturating_add(1);
+    pub fn process_details_down(&mut self, frame: &mut Frame) {
```

Comment:
This method calculates the height of an area and the number of content lines dynamically so we can decide when to stop scrolling.